waltz/src/code_block.rs

```diff
@@ -4,42 +4,50 @@ use std::io::Write;
 use std::path::Path;
 
 use errors::*;
+use code_flags::CodeFlags;
 
 /// Markdown code block
 ///
 /// This is the 'final' representation, with the filename and code already
 /// extracted.
 #[derive(Debug, Clone)]
 pub struct CodeBlock {
-    filename: Option<String>,
+    flags: Option<CodeFlags>,
     content: String,
 }
 
 impl Default for CodeBlock {
     fn default() -> Self {
-        CodeBlock { filename: None, content: String::new() }
+        CodeBlock { flags: None, content: String::new() }
     }
 }
 
 impl CodeBlock {
     /// Set the code block's filename
-    pub fn set_filename(&mut self, filename: String) {
-        self.filename = if filename.is_empty() { None } else { Some(filename) };
+    pub(crate) fn set_flags(&mut self, flags: CodeFlags) {
+        self.flags = Some(flags);
     }
 
-    /// Does the code block has a (non-empty) filename?
+    /// Does the code block have a (non-empty) filename?
     pub fn has_filename(&self) -> bool {
-        match self.filename {
-            Some(ref f) if !f.is_empty() => true,
-            _ => false,
+        self.filename().is_some()
+    }
+
+    /// Get the filename if it exists
+    pub fn filename(&self) -> Option<String> {
```

Comment:
Hm, it could be, but I'm printing this in a few places, which would not be as smooth when using Path. Does Path give me anything other than better semantic meaning? Every String is also a valid Path, right? (On unix at least)